modules/currency-l1/src/main/scala/org/tessellation/currency/l1/http/DataApplicationRoutes.scala

```diff
@@ -47,64 +52,135 @@ final case class DataApplicationRoutes[F[_]: Async: Hasher: SecurityProvider: L1
 
   protected val prefixPath: InternalUrlPrefix = "/"
 
-  private def validate(update: Signed[DataUpdate])(onValid: F[Response[F]]): F[Response[F]] =
+  private def validateDataUpdate(unsignedUpdate: DataUpdate)(onValid: F[Response[F]]): F[Response[F]] =
+    dataApplication.validateUpdate(unsignedUpdate).flatMap {
+      case Valid(_)   => onValid
+      case Invalid(e) => BadRequest(InvalidDataUpdate(e.toString).toApplicationError)
+    }
+
+  private def validateDataTransactions(
+    updates: DataTransactions
+  )(onValid: F[Response[F]]): F[Response[F]] =
     lastGlobalSnapshotStorage.getOrdinal.flatMap {
-      _.traverse(ord => (dataApplication.validateFee(ord)(update), dataApplication.validateUpdate(update.value)).mapN(_ |+| _))
-    }.flatMap {
-      case None             => InternalServerError(GL0SnapshotOrdinalUnavailable.toApplicationError)
-      case Some(Invalid(e)) => BadRequest(InvalidDataUpdate(e.toString).toApplicationError)
-      case Some(Valid(_))   => onValid
+      case Some(ordinal) =>
+        nodeContext.getLastCurrencySnapshotCombined.map(_.map(_._2.balances)).flatMap {
```

Comment:
Please don't use `_1` `_2`. It's really hard to read it without a context. It's better to use it like that: `.map { case (_, snapshotInfo) => snapshotInfo.balances }`

Reply:
👍🏼